pkg/controller/plan/adapter/openstack/builder.go

```diff
@@ -1113,9 +1113,9 @@ func (r *Builder) persistentVolumeClaimWithSourceRef(image model.Image, storageC
 	if originalVolumeDiskId, ok := image.Properties["forklift_original_volume_id"]; ok {
 		annotations[AnnImportDiskId] = originalVolumeDiskId.(string)
 		r.Log.Info("the image comes from a volume", "volumeID", originalVolumeDiskId)
-	} else {
-		annotations[AnnImportDiskId] = image.ID
-		r.Log.Info("the image comes from a vm snapshot", "imageID", image.ID)
+	} else if originalImageId, ok := image.Properties["forklift_original_image_id"]; ok {
+		annotations[AnnImportDiskId] = originalImageId.(string)
+		r.Log.Info("the image comes from a vm snapshot", "imageID", originalImageId)
 	}
```

Comment:
this code suggests that there might be another case where both properties don't exist even though it wouldn't make sense, right? maybe adding an `else` with a log?

Reply:
while on it, can you please replace 
`accessModes, volumeMode, err = r.getVolumeAndAccessMode(storageClassName)`
above with:
`accessModes, volumeMode, err := r.getVolumeAndAccessMode(storageClassName)`